File: CulinarySite/ServiceLayer/CulinaryChannelService.cs
using System.Collections.Generic;
using Database;
using Repositories;
using ServiceLayer.ViewModels.CulinaryChannel;

namespace ServiceLayer
{
    public class CulinaryChannelService : ICulinaryChannelService
    {
        private readonly IReadOnlyGenericRepository<CulinaryChannel> culinaryChannelReadOnlyRepository;
        private readonly IWriteGenericRepository<CulinaryChannel> culinaryChannelWriteRepository;
        public CulinaryChannelService(
            IReadOnlyGenericRepository<CulinaryChannel> culinaryChannelReadOnlyRepository,
            IWriteGenericRepository<CulinaryChannel> culinaryChannelWriteRepository)
        {
            this.culinaryChannelReadOnlyRepository = culinaryChannelReadOnlyRepository;
            this.culinaryChannelWriteRepository = culinaryChannelWriteRepository;
        }

        public void CreateCulinaryChannel(CreateCulinaryChannelModel createCulinaryChannelModel)
        {
            var culinaryChannel = new CulinaryChannel
            {
                Name = createCulinaryChannelModel.Name,
                Content = createCulinaryChannelModel.Content
            };

            this.culinaryChannelWriteRepository.Create(culinaryChannel);
            this.culinaryChannelWriteRepository.Save();
        }

        public void UpdateCulinaryChannel(UpdateCulinaryChannelModel updateCulinaryChannelModel)
        {
            var culinaryChannel = new CulinaryChannel
            {
                Id = updateCulinaryChannelModel.CulinaryChannelId,
                Name = updateCulinaryChannelModel.Name,
                Content = updateCulinaryChannelModel.Content
            };
            this.culinaryChannelWriteRepository.Update(culinaryChannel);
            this.culinaryChannelWriteRepository.Save();
        }

        public void DeleteCulinaryChannel(int id)
        {
            this.culinaryChannelWriteRepository.Delete(id);
            this.culinaryChannelWriteRepository.Save();
        }

        public IEnumerable<CulinaryChannelListModel> GetCulinaryChannelList(bool withRelated)
        {
            IEnumerable<CulinaryChannel> culinaryChannels;
            List<CulinaryChannelListModel> culinaryChannelListModels = new List<CulinaryChannelListModel>();
            if (withRelated)
            {
                culinaryChannels = this.culinaryChannelReadOnlyRepository.GetItemListWithInclude();
                foreach (var culinaryChannel in culinaryChannels)
                {
                    culinaryChannelListModels.Add(new CulinaryChannelListModel
                    {
                        CulinaryChannelId = culinaryChannel.Id,
                        Name = culinaryChannel.Name,
                        Content = culinaryChannel.Content
                    });
                }
                return culinaryChannelListModels;
            }
            culinaryChannels = this.culinaryChannelReadOnlyRepository.GetItemList();
            foreach (var culinaryChannel in culinaryChannels)
            {
                culinaryChannelListModels.Add(new CulinaryChannelListModel
                {
                    CulinaryChannelId = culinaryChannel.Id,
                    Name = culinaryChannel.Name,
                    Content = culinaryChannel.Content
                });
            }
            return culinaryChannelListModels;
        }

        public CulinaryChannelDetailModel GetCulinaryChannel(int id, bool withRelated)
        {
            var culinaryChannel = new CulinaryChannel();
            CulinaryChannelDetailModel culinaryChannelDetailModel = new CulinaryChannelDetailModel();

            if (withRelated)
            {
                culinaryChannel = this.culinaryChannelReadOnlyRepository.GetItemWithInclude(
                x => x.Id == id);
                culinaryChannelDetailModel = new CulinaryChannelDetailModel
                {
                    CulinaryChannelId = culinaryChannel.Id,
                    Name = culinaryChannel.Name,
                    Content = culinaryChannel.Content
                };
                return culinaryChannelDetailModel;
            }
            culinaryChannel = this.culinaryChannelReadOnlyRepository.GetItem(id);
            culinaryChannelDetailModel = new CulinaryChannelDetailModel
            {
                CulinaryChannelId = culinaryChannel.Id,
                Name = culinaryChannel.Name,
                Content = culinaryChannel.Content
            };
            return culinaryChannelDetailModel;
        }
    }
}

File: CulinarySiteApp/src/app/services/culinary-channel.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ICulinaryChannelListModel } from '../interfaces/culinary-channel/culinary-channel-list-model.interface';
import { ICulinaryChannelDetail } from '../interfaces/culinary-channel/culinary-channel-detail.interface';
import { CreateCulinaryChannelModel } from '../viewmodels/culinary-channel/create-culinary-channel-model.class';
import { UpdateCulinaryChannelModel } from '../viewmodels/culinary-channel/update-culinary-channel-model.class';
import { ICulinaryChannelModel } from '../interfaces/culinary-channel/culinary-channel.interface';

@Injectable()
export class CulinaryChannelService {
  public url: string = '/api/culinarychannel';

  constructor(private http: HttpClient) {}

  public getCulinaryChannelDetailList(
    withRelated: boolean
  ): Observable<ICulinaryChannelListModel[]> {
    return this.http.get<ICulinaryChannelListModel[]>(
      this.url + '/' + withRelated
    );
  }

  public getCulinaryChannelList(): Observable<ICulinaryChannelModel[]> {
    return this.http.get<ICulinaryChannelModel[]>(this.url);
  }

  public getCulinaryChannel(
    id: number,
    withRelated: boolean
  ): Observable<ICulinaryChannelDetail> {
    return this.http.get<ICulinaryChannelDetail>(
      this.url + '/' + id + '/' + withRelated
    );
  }

  public createCulinaryChannel(
    createCulinaryChannelModel: CreateCulinaryChannelModel
  ): Observable<void> {
    return this.http.post<void>(this.url, createCulinaryChannelModel);
  }

  public updateCulinaryChannel(
    updateCulinaryChannelModel: UpdateCulinaryChannelModel
  ): Observable<void> {
    return this.http.put<void>(this.url, updateCulinaryChannelModel);
  }

  public deleteCulinaryChannel(id: number): Observable<void> {
    return this.http.delete<void>(this.url + '/' + id);
  }
}
